File: component-acgschedule/src/main/java/com/rabtman/acgschedule/di/module/ScheduleNewModule.java
package com.rabtman.acgschedule.di.module;

import com.rabtman.acgschedule.mvp.contract.ScheduleNewContract;
import com.rabtman.acgschedule.mvp.model.ScheduleNewModel;
import com.rabtman.common.di.scope.ActivityScope;
import dagger.Module;
import dagger.Provides;

/**
 * @author Rabtman
 */
@Module
public class ScheduleNewModule {

  private ScheduleNewContract.View view;

  public ScheduleNewModule(ScheduleNewContract.View view) {
    this.view = view;
  }

  @ActivityScope
  @Provides
  ScheduleNewContract.View providerScheduleNewView() {
    return this.view;
  }

  @ActivityScope
  @Provides
  ScheduleNewContract.Model providerScheduleNewModel(ScheduleNewModel model) {
    return model;
  }
}

File: component-acgschedule/src/main/java/com/rabtman/acgschedule/di/module/ScheduleNewModule.kt
package com.rabtman.acgschedule.di.module

import com.rabtman.acgschedule.mvp.contract.ScheduleNewContract
import com.rabtman.acgschedule.mvp.model.ScheduleNewModel
import com.rabtman.common.di.scope.ActivityScope
import dagger.Module
import dagger.Provides

/**
 * @author Rabtman
 */
@Module
class ScheduleNewModule(private val view: ScheduleNewContract.View) {
    @ActivityScope
    @Provides
    fun providerScheduleNewView(): ScheduleNewContract.View {
        return view
    }

    @ActivityScope
    @Provides
    fun providerScheduleNewModel(model: ScheduleNewModel): ScheduleNewContract.Model {
        return model
    }

}